apps/murphyusagas/murphy_usa_gas.star

```diff
@@ -0,0 +1,285 @@
+API_STATION_SEARCH_RANGE = 20
+API_STATION_CACHE_KEY = "stations&lat={}&lng={}"
+API_STATION_LIST_TTL = 86400
+API_STATION_DETAILS_TTL = 3600
+DEBUG = False
+
+# #######################################################
+# #####           Where all the magic happens      ######
+# #######################################################
+def main(config):
+    gas_data = get_gas_data(config)
+
+    labels, prices = get_price_display(gas_data, config)
+
+    return render.Root(
+        max_age = 1800,
+        child = render.Column(
+            children = [
+                render.Marquee(
+                    width = 64,
+                    child = render.Text(gas_data["station"], color = "#0073A6"),
+                ),
+                render.Row(
+                    children = [
+                        render.Column(
+                            children = labels,
+                            cross_align = "start",
+                        ),
+                        render.Column(
+                            children = prices,
+                            cross_align = "end",
+                        ),
+                        render.Column(
+                            children = get_hours_display(gas_data),
+                            expanded = True,
+                            main_align = "center",
+                            cross_align = "end",
+                        ),
+                    ],
+                    expanded = True,
+                    main_align = "center",
+                    cross_align = "center",
+                ),
+            ],
+        ),
+    )
+
+# #######################################################
+# #####           Functions                        ######
+# #######################################################
+def get_schema():
+    price_colors = [
+        schema.Option(
+            display = "White",
+            value = "white",
+        ),
+        schema.Option(
+            display = "Red Gas, Green Diesel",
+            value = "red-green",
+        ),
+        schema.Option(
+            display = "Green Gas, Red Diesel",
+            value = "green-red",
+        ),
+    ]
+
+    return schema.Schema(
+        version = "1",
+        fields = [
+            schema.LocationBased(
+                id = "station_by_loc",
+                name = "Station",
+                desc = "A list of stations by location",
+                icon = "locationDot",
+                handler = get_stations,
+            ),
+            schema.Dropdown(
+                id = "price_color",
+                name = "Price Color",
+                desc = "Color scheme for price display",
+                icon = "palette",
+                default = DEFAULT_CONFIG["price_color"],
+                options = price_colors,
+            ),
+        ],
+    )
+
+def get_stations(location):
+    loc = json.decode(location) if location else json.decode(str(DEFAULT_LOCATION))
+    lat = loc["lat"]
```

Comment:
We'll need the lat/lng to be truncated a bit to avoid leaking users' exact locations to the API provider. Check out [this post](https://gis.stackexchange.com/questions/8650/measuring-accuracy-of-latitude-and-longitude) for some insight into how many decimals to keep.

Reply:
I'll get it done.  Thanks for the heads up.

Reply:
got it - down to 2 decimals.